File: julia/robot-name/robot-name.jl
const used = Set{String}()

mutable struct Robot
    rname::String
    Robot() = new(generate_name())
 end

function generate_name()::String
    newname = ""
    while true
        newname = join(rand('A':'Z', 2)) * join(rand(0:9, 3))
        if (!in(newname, used))
            push!(used, newname)
            break
        end
    end
    newname
end

function reset!(instance::Robot)
    instance.rname = generate_name()
    instance
end

function name(instance::Robot)
    instance.rname
end

File: python/robot-name/robot_name.py
import random
import string


def generate_name():
    return "".join(random.choices(string.ascii_uppercase, k=2)) + str(random.randint(100, 999))


class Robot:
    def __init__(self):
        self.name = generate_name()
        self._used_names = {self.name}

    def reset(self):
        while self.name in self._used_names:
            self.name = generate_name()
        self._used_names.add(self.name)
